supplier/tests/test_supplier_page.py

```diff
@@ -39,3 +91,66 @@ def test_supplier_check_false(non_supplier_mock):
 def test_supplier_check_true(saved_supplier0):
     assert saved_supplier0 in Supplier.objects.all()
     assert get_supplier(saved_supplier0) == saved_supplier0
+
+
+@pytest.mark.django_db()
+def test_receive_product_return_supplier_product(request_searched_product_mock,
+                                                 saved_supplier0,
+                                                 saved_supplier_product0,
+                                                 saved_supplier_product1,
+                                                 searched_product_string):
+    assert saved_supplier_product0.user_name == saved_supplier0
+    assert saved_supplier_product1.user_name == saved_supplier0
+    request_searched_product_mock.GET[searched_product_string] = saved_supplier_product0.qr_code.product_name
+    assert receive_product(request_searched_product_mock, saved_supplier0).first() == saved_supplier_product0
+
+
+@pytest.mark.django_db()
+def test_receive_product_return_none(request_searched_product_mock,
+                                     saved_supplier0,
+                                     searched_product_string):
+
+    request_searched_product_mock.GET[searched_product_string] = 'Fizzibublach'
+    assert receive_product(request_searched_product_mock, saved_supplier0) is None
+
+
+@pytest.mark.django_db()
+def test_searched_product_with_data(get_request_with_data,
+                                    render_mock,
+                                    saved_supplier_product0):
+    response = action_search_product(get_request_with_data, saved_supplier_product0)
+    assert response == render_mock.return_value
+
+    data = {'supplier_products': saved_supplier_product0}
+    assert render_mock.call_args_list == [
+        call(get_request_with_data, 'supplier/suppliers.html', data)
+    ]
+
+
+@pytest.mark.django_db()
+def test_searched_product_without_data(get_request_with_data,
+                                       render_mock,
+                                       messages_mock):
+    response = action_search_product(get_request_with_data, None)
+    assert response == render_mock.return_value
+    messages_mock.info.assert_called_with(
+        get_request_with_data, 'Product Not Found'
+    )
+    assert render_mock.call_args_list == [
+        call(get_request_with_data, 'supplier/suppliers.html')
+    ]
+
+
+@pytest.mark.django_db()
+def test_action_display_all_products(get_request,
+                                     saved_supplier0,
+                                     render_mock,
+                                     saved_supplier_product0,
+                                     saved_supplier_product1):
```

Comment:
Do we need these now?

Reply:
indeed, so saved supplier will have supplier products in db